daml-foundations/daml-ghc/daml-stdlib-src/DA/Internal/Template.daml

```diff
@@ -89,12 +89,30 @@ class Template c where
     internalArchive = magic @"archive"
 
 
+-- Deliberately not exported.
+data Consuming = PreConsuming  -- Archive before executing exercise body.
+               | PostConsuming -- Execute exercise body then archive.
+               | NonConsuming  -- Don't archive.
+
+-- | HIDE
+nonconsuming : NoEvent c e -> Consuming
+nonconsuming _ = NonConsuming
+
+-- | HIDE
+preconsuming : NoEvent c e -> Consuming
+preconsuming _ = PreConsuming
+
+-- | HIDE
+postconsuming : NoEvent c e -> Consuming
+postconsuming _ = PostConsuming
+
 class Template c => Choice c e r | c e -> r where
     -- | HIDE
-    consuming : NoEvent c e -> ChoiceType
-    -- must be a lambda since otherwise the default-dictionary method causes a runtime loop as the method
-    -- exists in the dictionary, takes the dictionary, and can't be eta expanded.
-    consuming _ = Consuming
+    consuming : NoEvent c e -> Consuming
+    -- Must be a lambda since otherwise the default-dictionary method
+    -- causes a runtime loop as the method exists in the dictionary,
+    -- takes the dictionary, and can't be eta expanded.
+    consuming = preconsuming
```

Comment:
You have a comment saying "Must be a Lambda", but code that isn't a Lambda. Sure this shouldn't be `consuming _ = Preconsuming`?

Reply:
Actually, the comment was pre-existing and I must admit, I never understood it. Evidently the current definition is OK? I'm inclined to delete the comment. @neil-da agree?

Reply:
Sure, delete the comment, if everything works

Reply:
✔️